File: FixMyCar/FixMyCar.Model/SearchObjects/ReservationSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FixMyCar.Model.SearchObjects
{
    public class ReservationSearchObject : BaseSearchObject
    {
        public string? CarRepairShopName { get; set; }
        public bool? Discount { get; set; }
        public string? State { get; set; }
        public string? Type { get; set; }
        public double? MinTotalAmount { get; set; }
        public double? MaxTotalAmount { get; set; }
        public bool? ClientOrder { get; set; }
        public DateTime? MinCreatedDate { get; set; }
        public DateTime? MaxCreatedDate { get; set; }
        public DateTime? MinReservationDate { get; set; }
        public DateTime? MaxReservationDate { get; set; }
        public DateTime? MinEstimatedCompletionDate { get; set; }
        public DateTime? MaxEstimatedCompletionDate { get; set; }
        public DateTime? MinCompletionDate { get; set; }
        public DateTime? MaxCompletionDate { get; set; }
    }
}

File: FixMyCar/FixMyCar.UI/fixmycar_client/lib/src/models/reservation/reservation_search_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation_search_object.g.dart';

@JsonSerializable()
class ReservationSearchObject {
  bool? discount;
  String? state;
  String? type;
  double? minTotalAmount;
  double? maxTotalAmount;
  bool? clientOrder;
  DateTime? minCreatedDate;
  DateTime? maxCreatedDate;
  DateTime? minReservationDate;
  DateTime? maxReservationDate;
  DateTime? minEstimatedCompletionDate;
  DateTime? maxEstimatedCompletionDate;
  DateTime? minCompletionDate;
  DateTime? maxCompletionDate;

  ReservationSearchObject.n({
    required this.minTotalAmount,
    required this.maxTotalAmount,
  });

  ReservationSearchObject(
      this.discount,
      this.state,
      this.type,
      this.minTotalAmount,
      this.maxTotalAmount,
      this.clientOrder,
      this.minCreatedDate,
      this.maxCreatedDate,
      this.minReservationDate,
      this.maxReservationDate,
      this.minEstimatedCompletionDate,
      this.maxEstimatedCompletionDate,
      this.minCompletionDate,
      this.maxCompletionDate);

  factory ReservationSearchObject.fromJson(Map<String, dynamic> json) =>
      _$ReservationSearchObjectFromJson(json);

  Map<String?, dynamic> toJson() => _$ReservationSearchObjectToJson(this);
}